### ISSUE TITLE
Add support for bcmath functions

### ISSUE THREAD

**Issue by BackEndTea** [thecodingmachine/safe#213]
BCMath functions have a hard time dealing with non numeric numbers, which can be a problem due to locales or scientific notations (which are numeric according to php).

I wrote an implementation for it already, but i  think it would be beneficial to add it to this package. It may be outside of the scope of this package, as it isnt' about wrong return types, but rather the results themselves being wrong.

**Comment by peter279k** [comment]
It's related to issue #202 :).

**Comment by Kharhamel** [comment]
I am going to try respond to both this and #202 since it is the same issue.

Safe isn't there to be a better standard library. 

There are lot of other projects that try to do this, written by people smarter and more dedicated than us. Safe is just there to automatically create wrappers around standard functions in order to throw exceptions on errors, with as little maintenance as possible, because editing the standard library is a rabbit hole we don't to fall into.
Features like yours or #202 or even some older issues, while not really hard to do by themselves, means we need to change the whole development philosophy behind this package. 

Right now, we are actually working to make Safe not needed in php8 by trying to integrate this philosophy of 'exception on errors' into the base library. We don't want to turn Safe into something bigger, quite the opposite actually.

In conclusion, I am sorry but no. It is simply out of scope.

**Comment by BackEndTea** [comment]
Thanks for the reply @Kharhamel, i figured it might be out of scope for this lib. But thank you for writing out why it is not within scope for this package. I hope the safe library will not be needed by php 8.0 😄
